sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -883,7 +889,7 @@ size_t gcoap_req_send2(const uint8_t *buf, size_t len,
             DEBUG("gcoap: can't wake up mbox; no timeout for msg\n");
         }
     }
-    if (res <= 0) {
+    if ((memo != NULL) && (res <= 0)) {
```

Comment:
Please rework this conditional so the DEBUG stmt on l.897 still is executed when memo is NULL.

Reply:
Done